lib/request.js

```diff
@@ -96,11 +96,11 @@ function processBuffer(data = []) {
     return data;
 }
 
-function handleResponse(statusCode, data) {
+function handleResponse(statusCode, data, response) {
```

Comment:
thanks for making this, looks good! just one small ask, have handleResponse take response rather than the statusCode
```suggestion
function handleResponse(response, data) {
    let statusCode = response.statusCode;
    ...


}
```